### PR TITLE
Adjust error codes and defines to FF 0.9

### PR DESCRIPTION
**Pull request by orenc17**
This change is [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/35)

### CODE REVIEW

**Comment by alzix** [comment]
Reviewed 18 of 19 files at r1, 1 of 1 files at r2.
Review status: all files reviewed, 7 unresolved discussions (waiting on @orenc17, @mikisch81, @danny4478, and @romkuz01)

---

*[spm/psa_defs.h, line 44 at r2](https://reviewable.io/reviews/kfnta/mbed-os/35#-LEzPwYF1VSOgr_KmDhj:-LEzPwYF1VSOgr_KmDhk:b-28nrmz) ([raw file](https://github.com/kfnta/mbed-os/blob/76f5e5b0468d32b2770dc43706a20ca4be29c541/spm/psa_defs.h#L44)):*
> ```objc
> #endif
> 
> #define PSA_FRAMEWORK_VERSION    (0x0009)
> ```

add doxygen comments for new defines

---

*[spm/psa_defs.h, line 65 at r2](https://reviewable.io/reviews/kfnta/mbed-os/35#-LEzQDt03aYnpxdxTitI:-LEzQDt03aYnpxdxTitJ:b6tvs1m) ([raw file](https://github.com/kfnta/mbed-os/blob/76f5e5b0468d32b2770dc43706a20ca4be29c541/spm/psa_defs.h#L65)):*
> ```objc
> #define PSA_MMIO_PERM_READ_WRITE                    0x000000003
> 
> #define PSA_DOORBELL (0x00000008u) // Mask for PSA_DOORBELL signal
> ```

doxygen comments please on everything in this file

---

*[spm/psa_defs.h, line 86 at r2](https://reviewable.io/reviews/kfnta/mbed-os/35#-LEzQQFS9yny4vkvhYli:-LEzQQFS9yny4vkvhYlj:b-b41c13) ([raw file](https://github.com/kfnta/mbed-os/blob/76f5e5b0468d32b2770dc43706a20ca4be29c541/spm/psa_defs.h#L86)):*
> ```objc
> typedef enum spm_msg_type {
>     PSA_IPC_MSG_TYPE_INVALID = 0,                     /**< Invalid message type */
>     PSA_IPC_MSG_TYPE_CONNECT = PSA_IPC_CONNECT,       /**< Sent by psa_connect() */
> ```

do we really need this type anymore?
if yes - it should be moved to internal header file. in addition the type is spm_* but the enum symbols are PSA_ :)
if we leaving it - let's use consistent naming

---

*[spm/spm_client.c, line 194 at r2](https://reviewable.io/reviews/kfnta/mbed-os/35#-LEzTH4Y869ImFbU5Cmh:-LEzTH4Y869ImFbU5Cmi:bt8xrwn) ([raw file](https://github.com/kfnta/mbed-os/blob/76f5e5b0468d32b2770dc43706a20ca4be29c541/spm/spm_client.c#L194)):*
> ```
> // TODO: Add DROP_CONNECTION handling
> ```

---

*[spm/spm_client.c, line 195 at r2](https://reviewable.io/reviews/kfnta/mbed-os/35#-LEzRFQK8ETsiDdqbDS1:-LEzRFQK8ETsiDdqbDS2:b-ix7xgs) ([raw file](https://github.com/kfnta/mbed-os/blob/76f5e5b0468d32b2770dc43706a20ca4be29c541/spm/spm_client.c#L195)):*
> ```C
>     spm_ipc_channel_t *channel = get_channel_from_handle(msg->channel);
> 
>     /* Prep code for total 4 IOvecs
> ```

please remove comment

---

*[spm/spm_server.h, line 69 at r2](https://reviewable.io/reviews/kfnta/mbed-os/35#-LEzRh3RD6J8BVhV3QP-:-LEzRh3RD6J8BVhV3QP0:be36mo1) ([raw file](https://github.com/kfnta/mbed-os/blob/76f5e5b0468d32b2770dc43706a20ca4be29c541/spm/spm_server.h#L69)):*
> ```objc
>  *
>  * @param[in] msg_handle Handle for the caller's message.
>  * @return Partition ID of the sender.
> ```

Caller partition ID

---

*[spm/spm_server.c, line 471 at r2](https://reviewable.io/reviews/kfnta/mbed-os/35#-LEzTs9v3tbiX5tPYW9w:-LEzTs9v3tbiX5tPYW9x:b-ch5otv) ([raw file](https://github.com/kfnta/mbed-os/blob/76f5e5b0468d32b2770dc43706a20ca4be29c541/spm/spm_server.c#L471)):*
> ```C
>         {
>             // TODO: Add DROP_CONNECTION handling
>             CHANNEL_STATE_ASSERT(active_channel->state, CHANNEL_STATE_ACTIVE);
> ```

The use of other reserved return codes in the range INT32_MIN+1 to INT32_MIN+127 is a fatal programming error.

---


*Comments from [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/35)*

**Comment by orenc17** [comment]
Review status: all files reviewed, 7 unresolved discussions (waiting on @alzix, @orenc17, @mikisch81, @danny4478, and @romkuz01)

---

*[spm/psa_defs.h, line 44 at r2](https://reviewable.io/reviews/kfnta/mbed-os/35#-LEzPwYF1VSOgr_KmDhj:-LFCgcBn0WAgCVV1RxPW:b-896fix) ([raw file](https://github.com/kfnta/mbed-os/blob/76f5e5b0468d32b2770dc43706a20ca4be29c541/spm/psa_defs.h#L44)):*
<details><summary><i>Previously, alzix (Alexander Zilberkant) wrote…</i></summary><blockquote>

add doxygen comments for new defines
</blockquote></details>

Done.

---

*[spm/psa_defs.h, line 65 at r2](https://reviewable.io/reviews/kfnta/mbed-os/35#-LEzQDt03aYnpxdxTitI:-LFCgcuvAOhg7UIq2iji:b-896fix) ([raw file](https://github.com/kfnta/mbed-os/blob/76f5e5b0468d32b2770dc43706a20ca4be29c541/spm/psa_defs.h#L65)):*
<details><summary><i>Previously, alzix (Alexander Zilberkant) wrote…</i></summary><blockquote>

doxygen comments please on everything in this file
</blockquote></details>

Done.

---

*[spm/psa_defs.h, line 86 at r2](https://reviewable.io/reviews/kfnta/mbed-os/35#-LEzQQFS9yny4vkvhYli:-LFCgdwaC38r70n7TkBn:b-yydsvx) ([raw file](https://github.com/kfnta/mbed-os/blob/76f5e5b0468d32b2770dc43706a20ca4be29c541/spm/psa_defs.h#L86)):*
<details><summary><i>Previously, alzix (Alexander Zilberkant) wrote…</i></summary><blockquote>

do we really need this type anymore?
if yes - it should be moved to internal header file. in addition the type is spm_* but the enum symbols are PSA_ :)
if we leaving it - let's use consistent naming
</blockquote></details>

we use this type in the partition main loop to determine which message we got 

---

*[spm/spm_client.c, line 194 at r2](https://reviewable.io/reviews/kfnta/mbed-os/35#-LEzTH4Y869ImFbU5Cmh:-LFCi5Ii-JU2xpKLxHhT:b-896fix) ([raw file](https://github.com/kfnta/mbed-os/blob/76f5e5b0468d32b2770dc43706a20ca4be29c541/spm/spm_client.c#L194)):*
<details><summary><i>Previously, alzix (Alexander Zilberkant) wrote…</i></summary><blockquote>

> ```
> // TODO: Add DROP_CONNECTION handling
> ```

</blockquote></details>

Done.

---

*[spm/spm_client.c, line 195 at r2](https://reviewable.io/reviews/kfnta/mbed-os/35#-LEzRFQK8ETsiDdqbDS1:-LFCi5XsBVq0VwWcmpgk:b-896fix) ([raw file](https://github.com/kfnta/mbed-os/blob/76f5e5b0468d32b2770dc43706a20ca4be29c541/spm/spm_client.c#L195)):*
<details><summary><i>Previously, alzix (Alexander Zilberkant) wrote…</i></summary><blockquote>

please remove comment
</blockquote></details>

Done.

---

*[spm/spm_server.h, line 69 at r2](https://reviewable.io/reviews/kfnta/mbed-os/35#-LEzRh3RD6J8BVhV3QP-:-LFCiDee3yIVnuTSzl0R:b-896fix) ([raw file](https://github.com/kfnta/mbed-os/blob/76f5e5b0468d32b2770dc43706a20ca4be29c541/spm/spm_server.h#L69)):*
<details><summary><i>Previously, alzix (Alexander Zilberkant) wrote…</i></summary><blockquote>

Caller partition ID
</blockquote></details>

Done.

---

*[spm/spm_server.c, line 471 at r2](https://reviewable.io/reviews/kfnta/mbed-os/35#-LEzTs9v3tbiX5tPYW9w:-LFCmnZZ7EXg3wJiHlUT:b-896fix) ([raw file](https://github.com/kfnta/mbed-os/blob/76f5e5b0468d32b2770dc43706a20ca4be29c541/spm/spm_server.c#L471)):*
<details><summary><i>Previously, alzix (Alexander Zilberkant) wrote…</i></summary><blockquote>

The use of other reserved return codes in the range INT32_MIN+1 to INT32_MIN+127 is a fatal programming error.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/35)*

**Comment by alzix** [comment]
Reviewed 4 of 4 files at r3.
Review status: all files reviewed, 4 unresolved discussions (waiting on @alzix, @orenc17, @mikisch81, @danny4478, and @romkuz01)

---

*[spm/psa_defs.h, line 86 at r2](https://reviewable.io/reviews/kfnta/mbed-os/35#-LEzQQFS9yny4vkvhYli:-LFCo86z12cvJftwzJBh:b-4hmkxa) ([raw file](https://github.com/kfnta/mbed-os/blob/76f5e5b0468d32b2770dc43706a20ca4be29c541/spm/psa_defs.h#L86)):*
<details><summary><i>Previously, orenc17 (Oren Cohen) wrote…</i></summary><blockquote>

we use this type in the partition main loop to determine which message we got 
</blockquote></details>

please remove this enum

---

*[spm/psa_defs.h, line 62 at r3](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFCnEtmEDBOHEa6yEiM:-LFCnEtn29B1GC5zdavO:b-q36te) ([raw file](https://github.com/kfnta/mbed-os/blob/16b2f1ca537e4eccb1450e566f0b7e7ae9b59644/spm/psa_defs.h#L62)):*
> ```objc
> #define PSA_MINOR_VERSION_POLICY_STRICT (1UL)  /**< Force minor version check during psa_connect().*/
> 
> #define PSA_MMIO_PERM_READ_ONLY   (0x000000001)
> ```

please move to internal.h

---

*[spm/psa_defs.h, line 106 at r3](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFCoK865L6hkCjx53Mz:-LFCoK865L6hkCjx53N-:b-7n7ky7) ([raw file](https://github.com/kfnta/mbed-os/blob/16b2f1ca537e4eccb1450e566f0b7e7ae9b59644/spm/psa_defs.h#L106)):*
> ```objc
>  */
> typedef struct psa_msg {
>     uint32_t type;                       /**< The message type, one of ::spm_msg_type.*/
> ```

when removing enum - fix the comment

---

*[spm/spm_server.c, line 471 at r2](https://reviewable.io/reviews/kfnta/mbed-os/35#-LEzTs9v3tbiX5tPYW9w:-LFCqvLFBMK5ywPZ01X6:bnxfczp) ([raw file](https://github.com/kfnta/mbed-os/blob/76f5e5b0468d32b2770dc43706a20ca4be29c541/spm/spm_server.c#L471)):*
<details><summary><i>Previously, orenc17 (Oren Cohen) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

not - done yet - return codes in the range INT32_MIN+1 to INT32_MIN+127 is a fatal programming error.

---


*Comments from [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/35)*

**Comment by alzix** [comment]
Reviewed 14 of 14 files at r4.
Review status: all files reviewed, 1 unresolved discussion (waiting on @orenc17, @mikisch81, @danny4478, and @romkuz01)

---

*[spm/spm_internal.h, line 42 at r4](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFCva8p4naJhoOmBEJn:-LFCva8p4naJhoOmBEJo:b-39hhdd) ([raw file](https://github.com/kfnta/mbed-os/blob/98c7a8c9959d7db18b15c7f39cbc1c852d275bd3/spm/spm_internal.h#L42)):*
> ```objc
> 
> #define PSA_IPC_MSG_TYPE_INVALID  (0)                    /**< Invalid message type */
> #define PSA_IPC_MSG_TYPE_MAX      (PSA_IPC_DISCONNECT)   /**< boundary checks for message type*/
> ```

please remove the second asterics and the `<`
`/**` is a doxygen  header start

---


*Comments from [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/35)*

**Comment by orenc17** [comment]
Review status: 18 of 20 files reviewed, 1 unresolved discussion (waiting on @alzix, @mikisch81, @danny4478, and @romkuz01)

---

*[spm/psa_defs.h, line 86 at r2](https://reviewable.io/reviews/kfnta/mbed-os/35#-LEzQQFS9yny4vkvhYli:-LFCuPEk2LuTtAh_qWjq:b-896fix) ([raw file](https://github.com/kfnta/mbed-os/blob/76f5e5b0468d32b2770dc43706a20ca4be29c541/spm/psa_defs.h#L86)):*
<details><summary><i>Previously, alzix (Alexander Zilberkant) wrote…</i></summary><blockquote>

please remove this enum
</blockquote></details>

Done.

---

*[spm/psa_defs.h, line 62 at r3](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFCnEtmEDBOHEa6yEiM:-LFCuNvN2Q_9NLp7Xvep:b-896fix) ([raw file](https://github.com/kfnta/mbed-os/blob/16b2f1ca537e4eccb1450e566f0b7e7ae9b59644/spm/psa_defs.h#L62)):*
<details><summary><i>Previously, alzix (Alexander Zilberkant) wrote…</i></summary><blockquote>

please move to internal.h
</blockquote></details>

Done.

---

*[spm/psa_defs.h, line 106 at r3](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFCoK865L6hkCjx53Mz:-LFCuQFPA9BC4biPIDaN:b-896fix) ([raw file](https://github.com/kfnta/mbed-os/blob/16b2f1ca537e4eccb1450e566f0b7e7ae9b59644/spm/psa_defs.h#L106)):*
<details><summary><i>Previously, alzix (Alexander Zilberkant) wrote…</i></summary><blockquote>

when removing enum - fix the comment
</blockquote></details>

Done.

---

*[spm/spm_internal.h, line 42 at r4](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFCva8p4naJhoOmBEJn:-LFCzvRz9zVSH9jqYo2Q:b-896fix) ([raw file](https://github.com/kfnta/mbed-os/blob/98c7a8c9959d7db18b15c7f39cbc1c852d275bd3/spm/spm_internal.h#L42)):*
<details><summary><i>Previously, alzix (Alexander Zilberkant) wrote…</i></summary><blockquote>

please remove the second asterics and the `<`
`/**` is a doxygen  header start
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/35)*

**Comment by alzix** [comment]
Reviewed 1 of 1 files at r7.
Review status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @mikisch81, @danny4478, and @romkuz01)

---



*Comments from [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/35)*

**Comment by mikisch81** [comment]
Review status: all files reviewed, 6 unresolved discussions (waiting on @orenc17, @danny4478, and @romkuz01)

---

*[spm/psa_defs.h, line 51 at r7](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFGnL028mmgTc86vaAt:-LFGnL028mmgTc86vaAu:b-ld4xxv) ([raw file](https://github.com/kfnta/mbed-os/blob/c01013aa1b1f38d5d031e6ac27d710fa11aff94b/spm/psa_defs.h#L51)):*
> ```objc
> #define PSA_NULL_HANDLE ((psa_handle_t)0)   /**< Denotes an invalid handle.*/
> 
> #define PSA_MAX_INVEC_LEN (3UL)  /**< Maximum number of psa_invec_t structures allowed for psa_call().*/
> ```

Can `PSA_MAX_INVEC_LEN` and `PSA_MAX_OUTVEC_LEN` defines be removed now?

---

*[spm/psa_defs.h, line 56 at r7](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFGnpS-4eDbKJf7cnDa:-LFGnpS-4eDbKJf7cnDb:b-m5kfvp) ([raw file](https://github.com/kfnta/mbed-os/blob/c01013aa1b1f38d5d031e6ac27d710fa11aff94b/spm/psa_defs.h#L56)):*
> ```objc
> #define PSA_MAX_IOVEC (4UL) /**< Maximum number of psa_invec_t and psa_outvec_t structures allowed for psa_call().*/
> 
> #define PSA_POLL  (0x00000000u) /**< Returns immediately even if none of the requested signals is asserted.*/
> ```

Is `u` postfix meaning unsigned? please stay consistent with UL for unsigned long, L for signed long.

---

*[spm/psa_defs.h, line 73 at r7](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFGoIpO3QeP_jVCry7o:-LFGoIpO3QeP_jVCry7p:bhmn8ia) ([raw file](https://github.com/kfnta/mbed-os/blob/c01013aa1b1f38d5d031e6ac27d710fa11aff94b/spm/psa_defs.h#L73)):*
> ```objc
> #define PSA_CONNECTION_REFUSED  (INT32_MIN + 1)    /**< The return value from psa_connect() if the RoT Service or SPM was unable to establish a connection.*/
> #define PSA_RESERVED_ERROR      (INT32_MIN + 127)  /**< Place holder for PSA error codes range.*/
> #define PSA_DROP_CONNECTION     (INT32_MIN)       /**< The result code in a call to psa_end() to indicate a non-recoverable error in the client.*/
> ```

Can you move this define before `PSA_CONNECTION_REFUSED` so it will be ascending?

---

*[spm/spm_client.c, line 214 at r7](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFGuw-GCdQm7HZM9B49:-LFGuw-GCdQm7HZM9B4A:bnm5pv9) ([raw file](https://github.com/kfnta/mbed-os/blob/c01013aa1b1f38d5d031e6ac27d710fa11aff94b/spm/spm_client.c#L214)):*
> ```C
>     }
> 
>     CHANNEL_STATE_ASSERT(channel->state, CHANNEL_STATE_IDLE);
> ```

Why is the type check removed?

---

*[spm/spm_internal.h, line 41 at r7](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFGv8fi9JHiqvv8fohV:-LFGv8fi9JHiqvv8fohW:b82ei4l) ([raw file](https://github.com/kfnta/mbed-os/blob/c01013aa1b1f38d5d031e6ac27d710fa11aff94b/spm/spm_internal.h#L41)):*
> ```objc
> #define SPM_COMPLETION_SEM_INITIAL_COUNT (0UL) /* Initial number of available tokens for a completion semaphore. */
> 
> #define PSA_IPC_MSG_TYPE_INVALID  (0)                    /* Invalid message type */
> ```

Maybe the name should align to the other types? `PSA_IPC_INVALID`, I am not too sure about this, what are your thoughts guys? @alzix @danny4478 

---

*[spm/spm_internal.h, line 80 at r7](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFGvjtnDLT-_NktvQWS:-LFGvjto-7UguG-lx1XT:bmqc4ju) ([raw file](https://github.com/kfnta/mbed-os/blob/c01013aa1b1f38d5d031e6ac27d710fa11aff94b/spm/spm_internal.h#L80)):*
> ```objc
>     psa_invec_t in_vec[PSA_MAX_INVEC_LEN]; /* IOvecs sent.*/
>     psa_outvec_t out_vec[PSA_MAX_OUTVEC_LEN]; /* IOvecs for response.*/
>     uint8_t type; /* The message type.*/
> ```

What happened to the `rc` field?

---


*Comments from [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/35)*

**Comment by orenc17** [comment]
Review status: all files reviewed, 6 unresolved discussions (waiting on @orenc17, @mikisch81, @danny4478, and @romkuz01)

---

*[spm/psa_defs.h, line 51 at r7](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFGnL028mmgTc86vaAt:-LFGxX_09SxbleWN6Tpe:b5ebero) ([raw file](https://github.com/kfnta/mbed-os/blob/c01013aa1b1f38d5d031e6ac27d710fa11aff94b/spm/psa_defs.h#L51)):*
<details><summary><i>Previously, mikisch81 (Michael Schwarcz) wrote…</i></summary><blockquote>

Can `PSA_MAX_INVEC_LEN` and `PSA_MAX_OUTVEC_LEN` defines be removed now?
</blockquote></details>

they are removed in the next PR

---

*[spm/psa_defs.h, line 56 at r7](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFGnpS-4eDbKJf7cnDa:-LFGz3fLCeEUA605RBgD:b-896fix) ([raw file](https://github.com/kfnta/mbed-os/blob/c01013aa1b1f38d5d031e6ac27d710fa11aff94b/spm/psa_defs.h#L56)):*
<details><summary><i>Previously, mikisch81 (Michael Schwarcz) wrote…</i></summary><blockquote>

Is `u` postfix meaning unsigned? please stay consistent with UL for unsigned long, L for signed long.
</blockquote></details>

Done.

---

*[spm/psa_defs.h, line 73 at r7](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFGoIpO3QeP_jVCry7o:-LFGy7cj5-8yX9r7AtFV:b-896fix) ([raw file](https://github.com/kfnta/mbed-os/blob/c01013aa1b1f38d5d031e6ac27d710fa11aff94b/spm/psa_defs.h#L73)):*
<details><summary><i>Previously, mikisch81 (Michael Schwarcz) wrote…</i></summary><blockquote>

Can you move this define before `PSA_CONNECTION_REFUSED` so it will be ascending?
</blockquote></details>

Done.

---

*[spm/spm_client.c, line 214 at r7](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFGuw-GCdQm7HZM9B49:-LFGyTfO7S1NlzH-LaRr:b-117x0i) ([raw file](https://github.com/kfnta/mbed-os/blob/c01013aa1b1f38d5d031e6ac27d710fa11aff94b/spm/spm_client.c#L214)):*
<details><summary><i>Previously, mikisch81 (Michael Schwarcz) wrote…</i></summary><blockquote>

Why is the type check removed?
</blockquote></details>

it has no use

---

*[spm/spm_internal.h, line 41 at r7](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFGv8fi9JHiqvv8fohV:-LFGynbX4vizjAx86zZW:bw9un32) ([raw file](https://github.com/kfnta/mbed-os/blob/c01013aa1b1f38d5d031e6ac27d710fa11aff94b/spm/spm_internal.h#L41)):*
<details><summary><i>Previously, mikisch81 (Michael Schwarcz) wrote…</i></summary><blockquote>

Maybe the name should align to the other types? `PSA_IPC_INVALID`, I am not too sure about this, what are your thoughts guys? @alzix @danny4478 
</blockquote></details>

it is a matter of alignment with the spec
both of these defines are made for our own boundary checks

---

*[spm/spm_internal.h, line 80 at r7](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFGvjtnDLT-_NktvQWS:-LFGz-PE0buSKTnRHvN3:b-oqsbqi) ([raw file](https://github.com/kfnta/mbed-os/blob/c01013aa1b1f38d5d031e6ac27d710fa11aff94b/spm/spm_internal.h#L80)):*
<details><summary><i>Previously, mikisch81 (Michael Schwarcz) wrote…</i></summary><blockquote>

What happened to the `rc` field?
</blockquote></details>

never used

---


*Comments from [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/35)*

File: spm/spm_server.c [hunk 1]
```diff
@@ -466,8 +466,12 @@ void psa_end(psa_handle_t msg_handle, psa_error_t retval)
             }
             break;
         }
-        case PSA_IPC_MSG_TYPE_CALL:
+        case PSA_IPC_CALL:
         {
+            if ((retval > PSA_DROP_CONNECTION) && (retval <= PSA_RESERVED_ERROR)) {
```
Comment (danny4478) [inline spm/spm_server.c]: Consider using PSA_RESERVED_ERROR_MIN  & PSA_RESERVED_ERROR_MAX for range check.

File: spm/psa_defs.h [hunk 2]
```diff
@@ -37,48 +37,43 @@ extern "C" {
...
 /* Error codes */
-#define PSA_INVALID_PARAMETERS        (-1L)
-#define PSA_INVALID_MEM               (-2L)
-#define PSA_CONNECTION_REFUSED_PERM   (-3L)
-#define PSA_CONNECTION_REFUSED_BUSY   (-4L)
-#define PSA_GENERIC_ERROR             (-5L)
-#define PSA_MEM_ALLOC_FAILED          (-6L)
-#define PSA_MSG_UNDELIVERED           (-7L)
+#define PSA_DROP_CONNECTION     (INT32_MIN)       /**< The result code in a call to psa_end() to indicate a non-recoverable error in the client.*/
+#define PSA_CONNECTION_REFUSED  (INT32_MIN + 1)    /**< The return value from psa_connect() if the RoT Service or SPM was unable to establish a connection.*/
```
Comment (danny4478) [inline spm/psa_defs.h]: Consider using PSA_RESERVED_ERROR_MIN  & PSA_RESERVED_ERROR_MAX for range check, and define using C file (if indeed used only there)

File: spm/spm_client.h [hunk 3]
```diff
@@ -65,8 +65,7 @@ uint32_t psa_version(uint32_t sid);
  * @param[in] sid          The Root of Trust Service ID.
  * @param[in] minor_version The minor version to be used for this connection.
  * @return A handle for the connection if greater than 0, else one of the following errors:@n
```
Comment (danny4478) [inline spm/spm_client.h]: There is only one error.
"...else, :" without "one of the following errors"

File: spm/spm_client.h [hunk 4]
```diff
@@ -82,11 +81,7 @@ psa_handle_t psa_connect(uint32_t sid, uint32_t minor_version);
  * @param[out] out_vec  Array of ::psa_outvec_t structures for optional Root of Trust Service response.
  * @param[in]  out_len  Number of ::psa_outvec_t structures in out_vec. (At most ::PSA_MAX_OUTVEC_LEN)
  * @return Positive numbers with an application-specific return code or negative numbers on error:@n
```
Comment (danny4478) [inline spm/spm_client.h]: Please re-write the @return lines to reflect that:
 >= 0 is application specific return code*
* < 0 is application specific error
* PSA_DROP_CONNECTION is a ....

**Comment by orenc17** [comment]
Review status: 14 of 20 files reviewed, 10 unresolved discussions (waiting on @alzix, @mikisch81, @danny4478, and @romkuz01)

---

*[spm/psa_defs.h, line 72 at r8](https://reviewable.io/reviews/kfnta/mbed-os/35#-LEtMoc5wqiu7jnYUJ5W-r8-72:-LFICmrI-A8Dz11A-qqx:b-896fix) ([raw file](https://github.com/kfnta/mbed-os/blob/fe6bd775fc4957d2545bf0b79f68bcfc3cf85db7/spm/psa_defs.h#L72)):*
<details><summary><i>Previously, danny4478 (Danny Shavit) wrote…</i></summary><blockquote>

Consider using PSA_RESERVED_ERROR_MIN  & PSA_RESERVED_ERROR_MAX for range check, and define using C file (if indeed used only there)
</blockquote></details>

Done.

---

*[spm/spm_client.h, line 67 at r8](https://reviewable.io/reviews/kfnta/mbed-os/35#-LEtMoc5wqiu7jnYUJ5Y-r8-67:-LFICna_9U527gOBNG6K:b-896fix) ([raw file](https://github.com/kfnta/mbed-os/blob/fe6bd775fc4957d2545bf0b79f68bcfc3cf85db7/spm/spm_client.h#L67)):*
<details><summary><i>Previously, danny4478 (Danny Shavit) wrote…</i></summary><blockquote>

There is only one error.
"...else, :" without "one of the following errors"

</blockquote></details>

Done.

---

*[spm/spm_server.c, line 471 at r8](https://reviewable.io/reviews/kfnta/mbed-os/35#-LEtMoc5wqiu7jnYUJ5Z-r8-471:-LFICoQADk_1gTcgywYE:b-896fix) ([raw file](https://github.com/kfnta/mbed-os/blob/fe6bd775fc4957d2545bf0b79f68bcfc3cf85db7/spm/spm_server.c#L471)):*
<details><summary><i>Previously, danny4478 (Danny Shavit) wrote…</i></summary><blockquote>

Consider using PSA_RESERVED_ERROR_MIN  & PSA_RESERVED_ERROR_MAX for range check.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/35)*

**Comment by alzix** [comment]
Reviewed 2 of 5 files at r8, 4 of 4 files at r9.
Review status: all files reviewed, 12 unresolved discussions (waiting on @mikisch81, @danny4478, @orenc17, and @romkuz01)

---

*[spm/spm_client.h, line 84 at r9](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFIDVRbBTTPbeDfig63:-LFIDVRbBTTPbeDfig64:b-nh3x1m) ([raw file](https://github.com/kfnta/mbed-os/blob/836faeeecbae4ae7bf8fbf00cc610b3c54917a82/spm/spm_client.h#L84)):*
> a ~every other~ negative number

a negative number

---

*[tools/spm/templates/psa_NAME_partition.h.tpl, line 61 at r9](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFILA-D2T5KtCikWC66:-LFILA-D2T5KtCikWC67:b-ol16di) ([raw file](https://github.com/kfnta/mbed-os/blob/836faeeecbae4ae7bf8fbf00cc610b3c54917a82/tools/spm/templates/psa_NAME_partition.h.tpl#L61)):*
> ```Smarty
> {% endif %}
> 
> #define {{partition.name|upper}}_WAIT_ANY_MSK (\
> ```

this macro is not longer needed

---


*Comments from [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/35)*

**Comment by alzix** [comment]
Dismissed @danny4478 and @mikisch81 from 8 discussions.
Review status: all files reviewed, 11 unresolved discussions (waiting on @alzix, @mikisch81, @danny4478, @orenc17, and @romkuz01)

---

*[spm/spm_internal.h, line 89 at r9](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFIRmU59GvxenAk-D3i:-LFIRmU59GvxenAk-D3j:b2hiaqq) ([raw file](https://github.com/kfnta/mbed-os/blob/836faeeecbae4ae7bf8fbf00cc610b3c54917a82/spm/spm_internal.h#L89)):*
> ```objc
>     psa_invec_t in_vec[PSA_MAX_INVEC_LEN]; /* IOvecs sent.*/
>     psa_outvec_t out_vec[PSA_MAX_OUTVEC_LEN]; /* IOvecs for response.*/
>     uint8_t type; /* The message type.*/
> ```

why do we need to store the type field?
we can easily access this info by `msg->channel->type`

---

*[spm/spm_server.c, line 319 at r9](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFIU4a1ApqHMpfk_ZMT:-LFIU4a1ApqHMpfk_ZMU:b-ki03ro) ([raw file](https://github.com/kfnta/mbed-os/blob/836faeeecbae4ae7bf8fbf00cc610b3c54917a82/spm/spm_server.c#L319)):*
> ```C
>         }
>         default:
>             SPM_PANIC("Unexpected message type %d\n", curr_channel->msg_type);
> ```

please change to assert

---

*[spm/spm_server.c, line 415 at r9](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFISJod8tHht_z4Muo1:-LFISJod8tHht_z4Muo2:b-xlmmds) ([raw file](https://github.com/kfnta/mbed-os/blob/836faeeecbae4ae7bf8fbf00cc610b3c54917a82/spm/spm_server.c#L415)):*
> ```C
> 
>     spm_active_msg_t *active_msg = get_msg_from_handle(msg_handle);
>     SPM_ASSERT((active_msg->type > PSA_IPC_MSG_TYPE_INVALID) &&
> ```

we can remove this check throughout whole SPM code and clean unused `PSA_IPC_MSG_TYPE_INVALID` and `PSA_IPC_MSG_TYPE_MAX`

---

*[spm/spm_server.c, line 483 at r9](https://reviewable.io/reviews/kfnta/mbed-os/35#-LFITSjh6-ELQeilsxLD:-LFITSjh6-ELQeilsxLE:b-trisug) ([raw file](https://github.com/kfnta/mbed-os/blob/836faeeecbae4ae7bf8fbf00cc610b3c54917a82/spm/spm_server.c#L483)):*
> ```C
>         }
>         default:
>             SPM_PANIC("Invalid message type %d\n", active_channel->msg_type);
> ```

this can actually be changed to ASSERT - as user can not directly cause this condition.
This check mainly validates our design assumptions.

---


*Comments from [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/35)*

**Comment by alzix** [comment]
Dismissed @danny4478 from a discussion.
Review status: all files reviewed, 10 unresolved discussions (waiting on @alzix, @mikisch81, @orenc17, and @romkuz01)

---



*Comments from [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/35)*

**Comment by alzix** [comment]
Reviewed 1 of 4 files at r10.
Dismissed @mikisch81 from a discussion.
Review status: 17 of 20 files reviewed, 7 unresolved discussions (waiting on @alzix, @orenc17, and @romkuz01)

---



*Comments from [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/35)*

**Comment by alzix** [comment]
:lgtm:

---







*Comments from [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/35#-:-LFIwHi7DRfGqeXu7AHH:bnfp4nl)*

**Comment by alzix** [comment]
Reviewed 3 of 4 files at r10.
Review status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @romkuz01)

---



*Comments from [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/35)*

**Pull request merged**